diff --git a/admin_users.go b/admin_users.go
index f8e26d9..3ac4cc3 100644
--- a/admin_users.go
+++ b/admin_users.go
@@ -42,6 +42,7 @@ type EditUserOption struct {
 	Admin            *bool  `json:"admin"`
 	AllowGitHook     *bool  `json:"allow_git_hook"`
 	AllowImportLocal *bool  `json:"allow_import_local"`
+	MaxRepoCreation  *int   `json:"max_repo_creation"`
 }
 
 func (c *Client) AdminEditUser(user string, opt EditUserOption) error {
