commit f487275d77abe1bcf0acc63ffa14ff13a1a55d2b
Author: roblabla <robinlambertz+dev@gmail.com>
Date:   Sun Mar 6 19:09:30 2016 +0100

    Add MaxRepoCreation to EditUser API

diff --git a/routers/api/v1/admin/user.go b/routers/api/v1/admin/user.go
index ce13804..fddfe0d 100644
--- a/routers/api/v1/admin/user.go
+++ b/routers/api/v1/admin/user.go
@@ -107,6 +107,9 @@ func EditUser(ctx *context.APIContext, form api.EditUserOption) {
 	if form.AllowImportLocal != nil {
 		u.AllowImportLocal = *form.AllowImportLocal
 	}
+	if form.MaxRepoCreation != nil {
+		u.MaxRepoCreation = *form.MaxRepoCreation
+	}
 
 	if err := models.UpdateUser(u); err != nil {
 		if models.IsErrEmailAlreadyUsed(err) {
