diff --git a/models/issue.go b/models/issue.go
index edc4668..753680b 100644
--- a/models/issue.go
+++ b/models/issue.go
@@ -33,8 +33,8 @@ var (
 // Issue represents an issue or pull request of repository.
 type Issue struct {
 	ID              int64 `xorm:"pk autoincr"`
-	RepoID          int64 `xorm:"INDEX"`
-	Index           int64 // Index in one repository.
+	RepoID          int64 `xorm:"INDEX unique(repoissue)"`
+	Index           int64 `xorm:"unique(repoissue)"` // Index in one repository.
 	Name            string
 	Repo            *Repository `xorm:"-"`
 	PosterID        int64
