commit 49a78f3b1849f8999e2449aebcc966ffd7810fdb
Author: roblabla <robinlambertz+dev@gmail.com>
Date:   Sun Mar 6 16:55:20 2016 +0100

    Add the AddOrgMembership API endpoint

diff --git a/routers/api/v1/api.go b/routers/api/v1/api.go
index 5050b8c..0ecf1d6 100644
--- a/routers/api/v1/api.go
+++ b/routers/api/v1/api.go
@@ -249,6 +249,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 		m.Group("/orgs/:orgname", func() {
 			m.Combo("").Get(org.Get).Patch(bind(api.EditOrgOption{}), org.Edit)
 			m.Combo("/teams").Get(org.ListTeams)
+			m.Put("/memberships/:username", bind(api.AddOrgMembershipOption{}), org.AddOrgMembership)
 		}, OrgAssignment(true))
 
 		m.Any("/*", func(ctx *context.Context) {
diff --git a/routers/api/v1/org/members.go b/routers/api/v1/org/members.go
new file mode 100644
index 0000000..bb5e634
--- /dev/null
+++ b/routers/api/v1/org/members.go
@@ -0,0 +1,47 @@
+// Copyright 2015 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package org
+
+import (
+	api "github.com/gogits/go-gogs-client"
+
+	"github.com/gogits/gogs/modules/context"
+	"github.com/gogits/gogs/routers/api/v1/convert"
+	"github.com/gogits/gogs/routers/api/v1/user"
+)
+
+func AddOrgMembership(ctx *context.APIContext, form api.AddOrgMembershipOption) {
+	org := user.GetUserByParamsName(ctx, ":orgname")
+	member := user.GetUserByParamsName(ctx, ":username")
+	if ctx.Written() {
+		return
+	}
+
+	if !org.IsOwnedBy(ctx.User.Id) {
+		ctx.Status(403)
+		return
+	}
+
+	if err := org.AddMember(member.Id); err != nil {
+		ctx.Error(500, "AddMember", err)
+		return
+	}
+	if form.Role == "admin" {
+		team, err := org.GetOwnerTeam();
+		if err != nil {
+			ctx.Error(500, "GetOwnerTeam", err)
+			return
+		}
+		if err := team.AddMember(member.Id); err != nil {
+			ctx.Error(500, "AddMember", err)
+			return
+		}
+	}
+	ret := map[string]interface{} {
+		"organization": convert.ToOrganization(org),
+		"user": convert.ToUser(member),
+	}
+	ctx.JSON(200, ret)
+}
